### PR TITLE
Prevent oversized transactions

### LINKED ISSUE #1573: Check transaction size
**Bug Description**
As reported in https://github.com/bitshares/bitshares-core/issues/374, transaction size hasn't been checked in consensus code, although we have a global chain parameter `maximum_transaction_size`.

https://github.com/bitshares/bitshares-core/blob/58c0c494a61c33e5a4ec2312084322f0014b4f5b/libraries/chain/include/graphene/chain/protocol/chain_parameters.hpp#L42

Note: before enabling this check with a hard fork, we should set the parameter to a fair value to avoid breaking things.

**Impacts**
Describe which portion(s) of BitShares Core may be impacted by this bug. Please tick at least one box.
- [ ] API (the application programming interface)
- [ ] Build (the build process or something prior to compiled code)
- [ ] CLI (the command line wallet)
- [ ] Deployment (the deployment process after building such as Docker, Travis, etc.)
- [ ] DEX (the Decentralized EXchange, market engine, etc.)
- [ ] P2P (the peer-to-peer network for transaction/block propagation)
- [ ] Performance (system or user efficiency, etc.)
- [x] Protocol (the blockchain logic, consensus, validation, etc.)
- [x] Security (the security of system or user data, etc.)
- [ ] UX (the User Experience)
- [ ] Other (please add below)

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Bug Report
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

### PR DESCRIPTION
**Pull request by jmjatlanta**
Fixes #1573

### CODE REVIEW

File: libraries/chain/protocol/transaction.cpp [hunk 1]
```diff
@@ -402,9 +407,17 @@ void precomputable_transaction::validate() const
 {
    if( _validated ) return;
    transaction::validate();
+   get_packed_size();
```
Comment (jmjatlanta) [inline libraries/chain/protocol/transaction.cpp]: Opinion: set  _packed_size here by calling transaction::get_packed_size() (slightly more efficient) or call precomputable_transaction::get_packed_size() (slightly more maintainable)?

Comment (pmconrad) [reply libraries/chain/protocol/transaction.cpp]: The call doesn't belong here at all. :-) `validate()` is supposed to check the validity of the TX, but you're only calling it for its side effect, i. e. to compute the size.
Better put it in https://github.com/bitshares/bitshares-core/blob/test-2.0.190212/libraries/chain/db_block.cpp#L747

Comment (jmjatlanta) [reply libraries/chain/protocol/transaction.cpp]: Who writes this garbage? Oh, it's me. :-(

**Comment by pmconrad** [comment]
Travis reported one failure in chain_test, worked upon restart. Strange.

**Pull request merged**